scene/resources/primitive_meshes.cpp

```diff
@@ -1382,7 +1382,7 @@ void QuadMesh::_create_mesh_array(Array &p_arr) const {
 		tangents.set(i * 4 + 0, 1.0);
 		tangents.set(i * 4 + 1, 0.0);
 		tangents.set(i * 4 + 2, 0.0);
-		tangents.set(i * 4 + 3, 1.0);
+		tangents.set(i * 4 + 3, -1.0);
```

Comment:
You are flipping the tangent also for a QuadMesh, which is the *only* builtin mesh that had correct tangents in vanilla godot. also, #16096 does not flip them: https://github.com/godotengine/godot/blob/3fe771b3eb7abf06c65261c8ab3d9d96e0bbabbe/scene/resources/primitive_meshes.cpp#L1315-L1318

I think your PR is wrong here, I will test this later.